src/api_server/src/http.rs

```diff
@@ -115,3 +118,99 @@ pub(crate) async fn attest(
 
     HttpResponse::Unauthorized().json(error_info)
 }
+
+/// GET /resource/{repository}/{type}/{tag}
+/// GET /resource/{type}/{tag}
+pub(crate) async fn resource(
+    request: HttpRequest,
+    repo_dir_path: web::Data<PathBuf>,
+    map: web::Data<SessionMap<'_>>,
+) -> HttpResponse {
+    let error_info = if let Some(cookie) = request.cookie(KBS_SESSION_ID) {
+        if let Some(locked_session) = map.sessions.read().await.get(cookie.value()) {
+            let session = locked_session.lock().await;
+
+            log::info!("Cookie {} request to get resource", session.id());
+
+            if !session.is_expired() {
```

Comment:
Shall we change this into `if session.is_expired()` to avoid the nested condition statement blocks?